### PR TITLE
Dtspo 15799 pipeline checks all envs

### PR DESCRIPTION
**Pull request by NickAzureDevops**
### Change description ###

Refactor pipeline to run sequentially on all enviroments: prevents changes which break just one env being merged to master which previously would be missed as PR checks only ran against sbox
Stages can still be picked individually on manual run by using the 'Stages to run' section on pipeline

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,342 +1,382 @@
...
-    - ithc
-    - preview
-    - demo
-    - sbox
-    - perftest
-    - aat
-    - ptlsbox
-    - ptl
-    - prod
+  - name: environment_components
```
Comment (thomast1906) [inline azure-pipelines.yml:54]: maybe something more meaning?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,342 +1,382 @@
...
-      value: "DCD-CFTAPPS-DEV"
-    ${{ if eq( parameters['env'], 'aat' ) }}: 
-      value: "DCD-CFTAPPS-STG"
-    ${{ if eq( parameters['env'], 'ptlsbox' ) }}: 
-      value: "DTS-CFTSBOX-INTSVC"
-    ${{ if eq( parameters['env'], 'ptl' ) }}: 
-      value: "DTS-CFTPTL-INTSVC"
-    ${{ if notIn( parameters['env'], 'aat', 'perftest', 'ptlsbox', 'preview', 'ptl') }}: 
-      value: "DCD-CFTAPPS-${{ parameters.env }}"
+    value: ubuntu-20.04
```
Comment (thomast1906) [inline azure-pipelines.yml:117]: could we not just use ubuntu-latest?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -1,342 +1,382 @@
...
   - name: tfInitSub
     value: "04d27a32-7a07-48b3-95b8-3c8691e1a263"
   - name: isMain
     value: $[eq(variables['Build.SourceBranch'], 'refs/heads/main')]
   - name: product
-    value: 'cft-platform'
+    value: cft-platform
```
Comment (thomast1906) [inline azure-pipelines.yml:123]: are all the cosmetic changes needed? Example:
'' to ""
"" to ''
'' to nothing

Comment (NickAzureDevops) [reply azure-pipelines.yml:123]: yes as before it was using double quote where it was needed and single quote too

File: azure-pipelines.yml (hunk 1 above)
Comment (NickAzureDevops) [reply azure-pipelines.yml:54]: All other pipeline in other repo used the same name so i left it there too

Comment (thomast1906) [reply azure-pipelines.yml:52]: not really same context I dont think, but can leave for now

**Review by thomast1906 — APPROVED** [review]
Check the terraform destroys/updates before applying 
- how we planning to test apply?

**Comment by NickAzureDevops** [comment]
> Check the terraform destroys/updates before applying
> 
> * how we planning to test apply?



> Check the terraform destroys/updates before applying
> 
> * how we planning to test apply?

All the changes are the same in main branch and spoke to Chirag and the preview is fine

**Review by danielwilsonkainos — APPROVED** [review]

**Pull request merged**